### PR TITLE
feat: add git hooks and lint staged [gh-516]

### PR DESCRIPTION
**Pull request by nahuelon**
I hereby confirm that I followed the code guidelines found at [engineering guidelines](https://www.notion.so/checkly/Engineering-Guidelines-a3a165a203a04dc1a112f0e26b2f2d3f)

## Affected Components
* [x] CLI
* [x] Create CLI
* [x] Test
* [x] Docs
* [x] Examples
* [x] Other

## Notes for the Reviewer
This PR modifies `cli` and `create-cli` packages adding `lint-staged`, `commitlint`, and `simple-git-hooks` hooks packages in the monorepo `package.json`. Some adjustments were applied to follow a monorepo project structure.

1. remove `examples/` from the workspaces lists
2. remove duplicated configuration for `eslint` (`package.json` section)
3. centralize linting and hooks by moving `eslint` and `simple-git-hooks` related packages from each workspace to the monorepo project
4. install the `simple-git-hooks` and `lint-staged` packages as `devDependencies` (and `prepare` script) within the monorepo project
5. adjust `CONTRIBUTING.md` docs.
6. fix linting small issues for `fixtures` that can compliance the linting rules, and disable `eslint` for files that are used to test 'syntax-errors'. Also, the are some fixtures files with some eslint rules disabled (like console.log, required, no-new
7. fix linting for `/examples` folder
8. adjust Github Actions to use monorepo project

Lot of files were modified (small changes) because I had to compliance all the linting rules before committing. This results in a very cleaned up source code.

All this changes are part of https://github.com/checkly/checkly-cli/issues/516 tasks.

### CODE REVIEW

File: packages/cli/constructs.js [hunk 1]
```diff
@@ -1 +1,3 @@
-module.exports = require('./dist').constructs
+import { constructs } from './dist'
```
Comment (umutuzgur) [inline packages/cli/constructs.js:1]: I think this has to stay the way it was, or?

File: commitlint.config.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  extends: ['@commitlint/config-conventional'],
```
[the remark this answers is not in the document]
Comment (umutuzgur) [reply commitlint.config.js:2]: Gotcha, my mistake

File: packages/cli/constructs.js (hunk 1 above)
Comment (nahuelon) [reply packages/cli/constructs.js:1]: I think we can use the `import`, but I'm trying it locally with different type of projects.

**Comment by nahuelon** [comment]
@umutuzgur , I adjusted all the previous linting configuration to use a monorepo approach and take advantage of its benefits. Now, dependency packages were removed from each project (`cli` and `create-cli`) and added to the monorepo project. I updated the PR description adding all changes the PR includes.

File: packages/cli/constructs.js (hunk 1 above)
Comment (umutuzgur) [reply packages/cli/constructs.js:1]: Because this is a module type

Comment (nahuelon) [reply packages/cli/constructs.js:1]: I'll revert this change for now.

File: packages/create-cli/src/actions/template.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ export async function copyTemplate ({ template, templatePath, targetDir }: CopyT
       downloadTemplateSpinner.text = chalk.red(`Couldn't find template "${template}"`)
       downloadTemplateSpinner.fail()
     } else {
-      console.error(e.message)
+      process.stderr.write(e.message)
```
Comment (umutuzgur) [inline packages/create-cli/src/actions/template.ts:25]: There is a difference between console and stderr.write. console operations always add a new line whereas stderr.write doesn't

**Review by umutuzgur — APPROVED** [review]
LGTM

**Pull request merged**